File: src/main/java/net/thedragonteam/mobsettings/client/gui/ConfigGui.java
/*
 * Copyright (c) TheDragonTeam 2016.
 */

package net.thedragonteam.mobsettings.client.gui;

import net.minecraft.client.gui.GuiScreen;
import net.minecraftforge.common.config.ConfigElement;
import net.minecraftforge.fml.client.config.GuiConfig;
import net.minecraftforge.fml.client.config.IConfigElement;
import net.thedragonteam.mobsettings.Reference;

import java.util.ArrayList;
import java.util.List;

import static net.thedragonteam.mobsettings.MobSettings.configuration;
import static net.thedragonteam.thedragonlib.util.TextHelper.localize;

public class ConfigGui extends GuiConfig {
    public ConfigGui(GuiScreen parentScreen) {
        super(parentScreen, getConfigElements(parentScreen), Reference.MODID, true, true,
                localize("gui." + Reference.MODID + ".config.title"));
    }

    private static List<IConfigElement> getConfigElements(GuiScreen parent) {
        List<IConfigElement> list = new ArrayList<>();

        /* adds sections declared in ConfigHandler. toLowerCase() is used because the configuration class automatically does this, so must we. */
        list.add(new ConfigElement(configuration.getCategory("Mobs".toLowerCase())));

        return list;
    }
}

File: src/main/kotlin/net/thedragonteam/mobsettings/client/gui/ConfigGui.kt
/*
 * Copyright (c) TheDragonTeam 2016.
 */

package net.thedragonteam.mobsettings.client.gui

import net.minecraft.client.gui.GuiScreen
import net.minecraftforge.common.config.ConfigElement
import net.minecraftforge.fml.client.config.GuiConfig
import net.minecraftforge.fml.client.config.IConfigElement
import net.thedragonteam.mobsettings.MobSettings.configuration
import net.thedragonteam.mobsettings.Reference
import net.thedragonteam.thedragonlib.util.TextHelper
import java.util.*

class ConfigGui(parentScreen: GuiScreen) : GuiConfig(parentScreen, getConfigElements(parentScreen), Reference.MODID, true, true, TextHelper.localize("gui." + Reference.MODID + ".config.title")) {
    companion object {
        private val list = ArrayList<IConfigElement>()

        private fun getConfigElements(parent: GuiScreen): List<IConfigElement> {
            list.add(ConfigElement(configuration.getCategory("mobs")))
            return list
        }
    }
}